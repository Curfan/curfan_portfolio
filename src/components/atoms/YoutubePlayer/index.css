.youtube-player {
	display: flex;
	position: relative;
	width: 100%;
	height: 100%;
	flex-direction: column;
	justify-content: center;
	align-items: center;

	&::before {
		content: '';
		position: absolute;
		top: 0;
		left: 0;
		width: 100%;
		height: 200px;
		background-image: linear-gradient(#1e222b, #1e222b, rgba(30, 34, 43, 0.5), rgba(30, 34, 43, 0));
		z-index: 1;
	}

	&::after {
		content: '';
		position: absolute;
		bottom: 0;
		left: 0;
		width: 100%;
		height: 200px;
		background-image: linear-gradient(rgba(30, 34, 43, 0), rgba(30, 34, 43, 0.5), #1e222b, #1e222b);
		z-index: 1;
	}

	& > iframe {
		width: 100%;
		height: 100%;
		pointer-events: none;
	}
}
