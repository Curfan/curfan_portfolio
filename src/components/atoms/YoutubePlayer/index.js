import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const YoutubePlayer = ({ className, youtubeId, title }) => (
	<div className={classnames(styles.youtubePlayer, className)}>
		<iframe
			title={title}
			src={`https://www.youtube.com/embed/${youtubeId}?modestbranding=1&autoplay=1&controls=0&fs=0&loop=1&rel=0&showinfo=0&disablekb=1&enablejsapi=1&mute=1&playlist=${youtubeId}`}
			frameBorder="0"
		/>
	</div>
);

export default YoutubePlayer;
