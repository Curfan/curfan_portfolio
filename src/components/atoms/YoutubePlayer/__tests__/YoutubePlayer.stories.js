import React from 'react';
import { storiesOf } from '@storybook/react';

import YoutubePlayer from 'components/atoms/YoutubePlayer';

const stories = storiesOf('atoms/YoutubePlayer', module);

stories.add('__interactive', () => <YoutubePlayer youtubeId="FneT5meY6s4" title="FneT5meY6s4" />);
