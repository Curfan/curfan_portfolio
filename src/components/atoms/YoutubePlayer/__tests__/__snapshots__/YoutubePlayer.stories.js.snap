// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="youtubePlayer"
  >
    <iframe
      frameborder="0"
      src="https://www.youtube.com/embed/FneT5meY6s4?modestbranding=1&autoplay=1&controls=0&fs=0&loop=1&rel=0&showinfo=0&disablekb=1&enablejsapi=1&mute=1&playlist=FneT5meY6s4"
      title="FneT5meY6s4"
    />
  </div>
</DocumentFragment>
`;
